### DISCUSSION TITLE
How to maintain absolute order in a multi-process situation？

### DISCUSSION THREAD

**Discussion by JanYork** [OptimalBits/bull#2778] [Q&A] [answered]
How to make the queue can keep consuming one after another regardless of the number of workers (processes), i.e., strongly consistent sequential consumption.

There can only be one task in progress, and the next task can only be started when the previous one completes or fails.

**Comment by roggervalf** [comment, marked as answer]
this is a feature we are exploring in bullmq package, but it's posponed until next breaking change. It won't be backported to bull so better to consider to migrate to use all our new features

**Reply by JanYork** [reply]
Has this been implemented in your new version? Is it open source and commercially available?

**Reply by JanYork** [reply]
> this is a feature we are exploring in bullmq package, but it's posponed until next breaking change. It won't be backported to bull so better to consider to migrate to use all our new features

However, I checked the documentation about bullmq, and it seems that it can't fulfill my requirement either, it seems that all of them can only fetch new messages sequentially, instead of waiting for the current one to be processed before starting to fetch the next one.
In a distributed/multiprocess environment.

**Reply by manast** [reply]
@JanYork as @roggervalf mentioned, it is coming in a newer major version as it implies a breaking change.

**Reply by JanYork** [reply]
Ok，Thank you very much for your reply.
